container/writer.go

```diff
@@ -96,6 +96,10 @@ func (avroWriter *Writer) WriteRecord(record AvroRecord) error {
 }
 
 func (avroWriter *Writer) Flush() error {
+	if !avroWriter.headerWritten {
+		return nil
+	}
```

Comment:
Ah yes, my bad.

It looks like I'd be able to put that `if` statement right before `block ...`, does that seem right? That way we close but not necessarily continue writing.